hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/OMSnapshotDirectoryMetrics.java

```diff
@@ -0,0 +1,399 @@
+
+    updateExecutor = Executors.newSingleThreadScheduledExecutor(r -> {
+      Thread t = new Thread(r, "OMSnapshotDirectoryMetricsUpdate");
+      t.setDaemon(true);
+      return t;
+    });
+
+    // Schedule periodic updates
+    updateTask = updateExecutor.scheduleWithFixedDelay(() -> {
+      try {
+        updateMetrics();
+        lastUpdateTime.set(System.currentTimeMillis());
+      } catch (Exception e) {
+        LOG.warn("Failed to update snapshot directory metrics", e);
+      }
+    }, 0, updateInterval, TimeUnit.MILLISECONDS);
+  }
+
+  /**
+   * Stops the periodic metrics update task.
+   */
+  public void stop() {
+    if (updateTask != null) {
+      updateTask.cancel(false); // Don't interrupt if running
+      updateTask = null;
+    }
+
+    if (updateExecutor != null) {
+      updateExecutor.shutdown();
+      try {
+        // Wait for any running updateMetrics() to complete (with timeout)
+        if (!updateExecutor.awaitTermination(30, TimeUnit.SECONDS)) {
+          LOG.warn("Metrics update executor did not terminate in time, forcing shutdown");
+          updateExecutor.shutdownNow();
+          // Wait a bit more for cancellation to take effect
+          if (!updateExecutor.awaitTermination(5, TimeUnit.SECONDS)) {
+            LOG.error("Metrics update executor did not terminate after force shutdown");
+          }
+        }
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+        updateExecutor.shutdownNow();
+      }
+      updateExecutor = null;
+    }
+  }
+
+  public void unRegister() {
+    stop();
+    MetricsSystem ms = DefaultMetricsSystem.instance();
+    ms.unregisterSource(SOURCE_NAME);
+  }
+
+  /**
+   * Internal class to store per-checkpoint metrics.
+   */
+  private static class CheckpointMetrics {
+    private final long size;
+    private final int sstFileCount;
+
+    CheckpointMetrics(long size, int sstFileCount) {
+      this.size = size;
+      this.sstFileCount = sstFileCount;
+    }
+
+    public long getSize() {
+      return size;
+    }
+
+    public int getSstFileCount() {
+      return sstFileCount;
+    }
+  }
+
+  private OMSnapshotDirectoryMetrics(OMMetadataManager metadataManager) {
+    this.metadataManager = metadataManager;
+  }
+
+  public static OMSnapshotDirectoryMetrics create(String parent,
+      OMMetadataManager metadataManager) {
+    MetricsSystem ms = DefaultMetricsSystem.instance();
+    return ms.register(SOURCE_NAME,
+        parent,
+        new OMSnapshotDirectoryMetrics(metadataManager));
+  }
+
+  /**
+   * Updates all metrics synchronously - both aggregate and per-checkpoint-directory.
+   */
+  @VisibleForTesting
+  void updateMetrics() throws IOException {
+    DBStore store = metadataManager.getStore();
+    if (!(store instanceof RDBStore)) {
+      LOG.debug("Store is not RDBStore, skipping snapshot directory metrics update");
+      resetMetrics();
+      return;
+    }
+
+    RDBStore rdbStore = (RDBStore) store;
```

Comment:
No need for type casting here you can directly call store.getSnapshotsParentDir()